### PR TITLE
Moving to typescript, removing redux, using out of box oidc solution

### PR DESCRIPTION
**Pull request by devinmatte**
Upgrading the boilerplate to be more ready for production use, without forcing the need for redux.

I think redux is overkill especially for new React projects, so I decided to replace our solution with a well made package that is feature ready for what we need.

Also typescript imo is just better to write, easier to read, and helps users catch problems way earlier than if it was solid js

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -3,10 +3,7 @@
         "browser": true,
         "es6": true
     },
-    "extends": [
-        "standard",
-        "plugin:react/recommended"
-    ],
+    "extends": ["plugin:prettier/recommended"],
```
Comment (stevenmirabito) [inline .eslintrc.json:6]: I think we should still extend a style guide. I like Airbnb but Standard is good too. The Prettier preset basically just enables Prettier to run as part of ESLint; it doesn't enforce any other standards on the code.

File: prettier.config.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
+    printWidth: 80,
+    tabWidth: 4,
+    singleQuote: true,
+    arrowParens: 'avoid',
+    trailingComma: 'none'
+};
```
Comment (stevenmirabito) [inline prettier.config.js:7]: Why not just use the standard Prettier style? `tabWidth: 4` is decidedly non-standard for JS, and the `trailingComma` is useful to make Git diffs less noisy. I don't think this boilerplate should be trying to enforce a non-standard code style.

File: src/components/App.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+const ProtectedHome = withOidcSecure(Home);
+
+class App extends Component {
+    render() {
+        return (
+            <Router>
+                <PageContainer>
+                    <Switch>
+                        <Route exact path="/" component={ProtectedHome} />
```
Comment (stevenmirabito) [inline src/components/App.tsx:15]: nit: just call `withOidcSecure` here to remove the instantiation of another variable. This pattern will just end with a huge list of variables at the top of the file and make this less readable.
```suggestion
                        <Route exact path="/" component={withOidcSecure(Home)} />
```

File: src/components/Home/index.ts [hunk 4]
```diff
@@ -0,0 +1,3 @@
+import Home from './Home';
+
+export default Home;
```
Comment (stevenmirabito) [inline src/components/Home/index.ts:3]: I don't see a need to have these components live in `<Component>.tsx` and then have a separate `index.ts` that just re-exports it. I'd just put the component code directly in `index.tsx`.

File: src/components/NavBar/NavBar.tsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    Navbar,
+    NavbarToggler,
+    NavItem
+} from 'reactstrap';
+import { NavLink } from 'react-router-dom';
+import Profile from './Profile';
+
+interface NavBarProps {}
+
+const NavBar: React.FunctionComponent<NavBarProps> = ({}) => {
```
Comment (stevenmirabito) [inline src/components/NavBar/NavBar.tsx:15]: If this component doesn't take any props, there's no need for an empty object de-structuring here. Also, while it's arguably here to guide in the right direction for future use cases, I don't see a need to provide an empty props interface.

File: src/components/NavBar/Profile.tsx [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+import {
+    DropdownItem,
+    DropdownMenu,
+    DropdownToggle,
+    UncontrolledDropdown
+} from 'reactstrap';
+
+interface ProfileProps {}
+
+const Profile: React.FunctionComponent<ProfileProps> = () => {
```
Comment (stevenmirabito) [inline src/components/NavBar/Profile.tsx:12]: Same as above. No need for an empty props interface.

File: src/components/NavBar/NavBar.tsx [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+    const [isOpen, setIsOpen] = React.useState<boolean>(false);
+
+    const toggle = () => {
+        setIsOpen(!isOpen);
+    };
```
Comment (stevenmirabito) [inline src/components/NavBar/NavBar.tsx:20]: Something to consider here would be to bring in `react-use`, which provides a `useToggle` among other nice hook utilities. This is fine, though, if the goal is to keep dependencies minimal.

File: src/components/NavBar/Profile.tsx [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+                    height={32}
+                />
+                {name} ({preferred_username})
+                <span className="caret" />
+            </DropdownToggle>
+            <DropdownMenu>
+                <DropdownItem>Dashboard</DropdownItem>
+                <DropdownItem>Settings</DropdownItem>
+                <DropdownItem divider />
+                <DropdownItem onClick={() => logout()}>Logout</DropdownItem>
```
Comment (stevenmirabito) [inline src/components/NavBar/Profile.tsx:45]: No need for the extra function here, just directly pass `logout`:
```suggestion
                <DropdownItem onClick={logout}>Logout</DropdownItem>
```

File: src/components/NavBar/Profile.tsx [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+interface ProfileProps {}
+
+const Profile: React.FunctionComponent<ProfileProps> = () => {
+    const { oidcUser, logout, events } = useReactOidc();
+
+    const [isOpen, setIsOpen] = React.useState<boolean>(false);
+
+    const toggle = () => {
+        setIsOpen(!isOpen);
+    };
```
Comment (stevenmirabito) [inline src/components/NavBar/Profile.tsx:19]: This doesn't seem to use this state/toggle.

File: src/components/Home/Home.tsx [hunk 10]
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+import InfoSpinner from '../InfoSpinner';
+
+const Home: React.FunctionComponent = () => {
+    return <InfoSpinner>Loading Content</InfoSpinner>;
```
Comment (stevenmirabito) [inline src/components/Home/Home.tsx:5]: To reduce confusion, this should probably render something static (perhaps `Welcome to the React Boilerplate!`) so it's clear the page is fully loaded.

File: package.json [hunk 11]
```diff
@@ -1,55 +1,60 @@
...
-  }
+    "name": "csh-react-boilerplate",
+    "version": "1.0.0",
+    "private": true,
+    "dependencies": {
+        "@axa-fr/react-oidc-context": "^3.1.5",
+        "@fortawesome/fontawesome-svg-core": "^1.2.31",
+        "@fortawesome/free-brands-svg-icons": "^5.15.0",
+        "@fortawesome/free-solid-svg-icons": "^5.15.0",
+        "@fortawesome/react-fontawesome": "^0.1.11",
```
Comment (stevenmirabito) [inline package.json:10]: We don't seem to use any iconography in the example app. Remove these from the default list of dependencies.

File: package.json [hunk 12]
```diff
@@ -1,55 +1,60 @@
...
+        "cross-fetch": "^3.0.6",
+        "csh-material-bootstrap": "^4.5.2",
```
Comment (stevenmirabito) [inline package.json:12]: Does this get imported anywhere? I think this _should_ be included as the default, but I don't see it.

File: package.json [hunk 13]
```diff
@@ -1,55 +1,60 @@
...
+    "name": "csh-react-boilerplate",
+    "version": "1.0.0",
+    "private": true,
+    "dependencies": {
+        "@axa-fr/react-oidc-context": "^3.1.5",
+        "@fortawesome/fontawesome-svg-core": "^1.2.31",
+        "@fortawesome/free-brands-svg-icons": "^5.15.0",
+        "@fortawesome/free-solid-svg-icons": "^5.15.0",
+        "@fortawesome/react-fontawesome": "^0.1.11",
+        "cross-fetch": "^3.0.6",
```
Comment (stevenmirabito) [inline package.json:11]: I don't see this used anywhere.

**Review by stevenmirabito — CHANGES_REQUESTED** [review]
Some comments. Also, consider making this a generator (e.g. `create-csh-app`) with [something like Sao](https://github.com/saojs/sao) to ease starting a new project.

File: prettier.config.js (hunk 2 above)
Comment (devinmatte) [reply prettier.config.js:7]: Fair. This is my personal preference but for the boilerplate we should probably stick with default prettier I suppose

File: .eslintrc.json (hunk 1 above)
Comment (devinmatte) [reply .eslintrc.json:6]: I could just bring back standard then

File: src/components/NavBar/Profile.tsx (hunk 8 above)
Comment (devinmatte) [reply src/components/NavBar/Profile.tsx:45]: I'm afraid we can't. The typing returned clashes in the ts compiler

File: .eslintrc.json (hunk 1 above)
Comment (devinmatte) [reply .eslintrc.json]: Turns out prettier does enforce with the eslint plugin, which is what I wanted. I got rid of my changes, but left prettier as the style config

File: package.json (hunk 12 above)
Comment (devinmatte) [reply package.json]: I don't see it either. We use themeswitcher so it may not get used at all

**Pull request merged**